File: src/main/java/com/cartrecipes/controller/dto/cart/CartDto.java
package com.cartrecipes.controller.dto.cart;

import com.cartrecipes.controller.dto.product.ProductDto;
import com.cartrecipes.controller.dto.recipe.ListingRecipeDto;

import java.util.List;

public record CartDto(Long id,
                      Integer totalInCents,
                      List<ProductDto> products,
                      List<ListingRecipeDto> recipes) {
}

File: src/main/kotlin/com/cartrecipes/controller/dto/cart/CartDto.kt
package com.cartrecipes.controller.dto.cart

import com.cartrecipes.controller.dto.product.ProductDto
import com.cartrecipes.controller.dto.recipe.ListingRecipeDto

class CartDto {
    var id: Long = 0
    var totalInCents: Int = 0
    var products: List<ProductDto> = emptyList()
    var recipes: List<ListingRecipeDto> = emptyList()
}